src/models/ProcessorOptions.ts

```diff
@@ -0,0 +1,6 @@
+export interface AsyncapiProcessorOptions {
+  path?: string
```

Comment:
The reality is that I don't fully understand the use case you mentioned in the PR description.
I'm concerned we are adding this here because of a particular use case on our side (AsyncAPI) but the cost of it is strongly coupling with AsyncAPI itself. 

Would you mind clarifying the use case and perhaps add a real example?

Reply:
Hmm, it is honestly a good point -  added a follow-up issue in the parser as well https://github.com/asyncapi/parser-js/issues/344

The example can be found in the follow-up issue. 

I think it is a difficult question, to answer whether we need to be this coupled, but do you see another approach? 🤔

Reply:
Ok, i understand now the purpose of this change (thanks to the example you set in the issue).

A small suggestion (if makes sense): I see that if we ever support any other input beside the path to an AsyncApi document (e.g. any dsl) this could also be needed (depending on the parser used).
would it make sense to just rename the `asyncapi` property on the option object to a more generic one such as `input` or similar.

If you don't see it make sense, forget about it.

Reply:
Maybe, but adding it as generic already without seeing the use-case for it, might be the wrong call? 🤔

Reply:
I dont think it is a wrong movement to just rename it for abstracting it from AsyncAPI but as mentioned, not strong opinion at this moment.

Reply:
I think it is a pre-optimization that might never be needed elsewhere. Let's keep it as is, for now, we can always reevaluate it for v1